truffle/src/main/java/org/jruby/truffle/core/module/ModuleNodes.java

```diff
@@ -398,7 +398,8 @@ private void createAccesor(DynamicObject module, String name) {
             final RubyNode sequence = Translator.sequence(getContext(), sourceSection.getSource(), rubySourceSection, Arrays.asList(checkArity, accessInstanceVariable));
             final RubyRootNode rootNode = new RubyRootNode(getContext(), sourceSection, null, sharedMethodInfo, sequence, false);
             final CallTarget callTarget = Truffle.getRuntime().createCallTarget(rootNode);
-            final InternalMethod method = new InternalMethod(sharedMethodInfo, accessorName, module, visibility, callTarget);
+            final boolean builtIn = !getContext().isRunning();
+            final InternalMethod method = new InternalMethod(sharedMethodInfo, accessorName, module, visibility, builtIn, callTarget);
```

Comment:
This logic could be part of addMethod or InternalMethod constructor to avoid duplication.